core/node/storage/pg_stream_store.go

```diff
@@ -171,6 +201,53 @@ func sqlForStream(sql string, streamId StreamId) string {
 	return sql
 }
 
+// isStreamMigrated checks the es table to see if the stream has been migrated to a fixed partition,
+// or if the data exists on it's own set of tables, according to the previous schema. If the
+// ignoreIfUnfound flag is set, a missing stream will be considered migrated and will not produce an
+// error.
+func (s *PostgresStreamStore) isStreamMigrated(
+	ctx context.Context,
+	tx pgx.Tx,
+	streamId StreamId,
+	ignoreIfUnfound bool,
+) (migrated bool, err error) {
+	if ignoreIfUnfound {
+		// Use the Query interface in order not to put the transaction into an error state if the row
+		// does not exist in the es table. The API of this method is to return en empty result for
+		// nonexistent streams.
+		esRows, err := tx.Query(ctx, "SELECT migrated FROM es WHERE stream_id = $1", streamId)
+		if err != nil {
+			return false, err
+		}
+		defer esRows.Close()
+
+		// If no such stream exists, consider the stream migrated.
```

Comment:
This logic is unclear: it what situation stream is not going to exist, but still checked for migration? (i.e. wouldn't caller of this function fail anyway if stream doesn't exist?)

Reply:
For sure! Honestly when reading this again I have no idea why I did that.